CoreFunctionality/src/au/gov/asd/tac/constellation/functionality/startup/ConstellationLAFSettings.java

```diff
@@ -0,0 +1,502 @@
+import javax.swing.JFrame;
+import javax.swing.Painter;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import org.openide.windows.WindowManager;
+
+/**
+ * This class contains a collection of separate UI settings to apply for a
+ * few different Look and Feels packaged with NetBeans 12.0
+ *
+ * Note: Since the code is based on NetBeans 12.0, it may require updating with
+ * newer versions NetBeans if the internal Look And Feel packages are changed.
+ *
+ * @author OrionsGuardian
+ */
+public class ConstellationLAFSettings {
+
+    private static final Logger LOGGER = Logger.getLogger(ConstellationLAFSettings.class.getName());
+    
+    private ConstellationLAFSettings() {
+        // Added this private constructor to hide the implicit public one
+    }
+    
+    public static void applyTabColorSettings() {
+        final JFrame mainframe = (JFrame) WindowManager.getDefault().getMainWindow();
+        final String currentLafName = UIManager.getLookAndFeel().getName().toUpperCase();
+        // Call the appropriate init method for the current Look And Feel
+        if (currentLafName.startsWith("WINDOWS")) {
+            initWindowsTabColors();
+        } else if (currentLafName.contains("NIMBUS")) {
+            initNimbusTabColors(currentLafName.contains("DARK"));
+        } else if (currentLafName.contains("FLAT")) {
+            initFlatLafTabColors(currentLafName.contains("DARK"));
+        } else if (currentLafName.contains("METAL")) {
+            initMetalTabColors(currentLafName.contains("DARK"));
+        }
+        if (mainframe != null) {
+            SwingUtilities.updateComponentTreeUI(mainframe);
+        }
+    }
+
+    /**
+     * Applies a Blue color theme for the main window tabs when using either the
+     * <b>Windows</b> or <b>Windows Classic</b> Look and Feel.
+     */
+    private static void initWindowsTabColors() {
+        // Fixed set of Blue shade colors
+        // appropriate for Windows LAF
+        final Color selectedUpperLightBlue = new Color(225, 235, 255);
+        final Color activeMidSectionBlue = new Color(160, 205, 255);
+        final Color selectedLowerDarkBlue = new Color(80, 160, 255);
+        final Color unselectedUpperGray = new Color(225, 225, 225);
+        final Color unselectedLowerGreyBlue = new Color(175, 185, 195);
+
+        try {
+            // To apply new tab colors to a Windows LAF
+            // we need to find the current TabDisplayerUI
+            // and force it to re-read from a revised set of colors
+            Class<?> displayerClass = (Class<?>) UIManager.getDefaults()
+                    .get("org.netbeans.swing.tabcontrol.plaf.Windows8VectorViewTabDisplayerUI");
+            if (displayerClass != null) {
+                final Class<?> tabDisplayer = displayerClass.getSuperclass();
+
+                if (tabDisplayer == null) {
+                    LOGGER.info(" >>> Unable to apply Tab coloring to Windows LAF : no superclass for displayerClass :");
```

Comment:
Could you possibly change the to either warning or severe?

Reply:
Done